crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -411,6 +411,7 @@ pub struct CheckpointBuilder {
     exit: watch::Receiver<()>,
     metrics: Arc<CheckpointMetrics>,
     max_transactions_per_checkpoint: usize,
+    max_checkpoint_size: usize,
```

Comment:
I almost wonder if now we don't need the `max_transactions_per_checkpoint` dimension...But we can figure that out later, its not important now.

Reply:
I figured it wouldn't hurt, it might take too long to process an unlimited # of tiny tx